src/lib/reasoners/bitv_rel.ml

```diff
@@ -164,7 +164,10 @@ end = struct
     | Bor -> Fmt.pf ppf "bvor"
     | Bxor -> Fmt.pf ppf "bvxor"
 
-  let equal_binop : binop -> binop -> bool = Stdlib.(=)
+  let equal_binop op1 op2 =
+    match op1, op2 with
+    | Band, Band | Bor, Bor | Bxor, Bxor -> true
+    | _ -> false
```

Comment:
This is not a good way to write such a comparison function as it will silently fail if we add new constructors to the `binop` type. Polymorphic comparison is OK on enumerated types, which `binop` is (and is intended to stay). I'd rather add a comment mentioning this on `binop`.

Reply:
I understand. What I am saying is that this should not happen: this particular type is designed to be an enumerated type, and it should never contain Dolmen identifiers. Hence, documenting that fact is a better fix

(Furthermore, I don't think the change is currently "Dolmen identifier proof" because I don't think the `hash` function below will work properly with Dolmen identifiers either)

Reply:
I think removing the polymorphic comparison is a better fix. 

I know the hash isn't correct.